app/controllers/api/v1/admin/role_permissions_controller.rb

```diff
@@ -31,6 +32,19 @@ def update
         def role_params
           params.require(:role).permit(:role_id, :name, :value)
         end
+
+        def default_room
+          return unless role_params[:name] == 'CreateRoom' && role_params[:value] == true
+
+          # find users with role id and create a room for them if their room count is <= 0
+          User.where(role_id: role_params[:role_id]).find_in_batches do |group|
```

Comment:
The default batch size is 1000. Should we lower it to like 500?

Reply:
default is fine - but you're doing the find in batches and then doing 1 create for each user in that find. You can instead do 1 create call per batch

Reply:
We may also want to add a check to make sure we're only creating rooms for users with no room

Reply:
Are you sure about the 1 create call per batch? I know you can `update` a batch in one call, but to create `Rooms` for `Users` with one call per batch i couldn't find anything. I can't find a way to pass a list of users into one create call. We also need to be able to access the user's `name` and `id` for each `room` being created.

There was already a check to make sure the user had no room in the  loop. I made it better and changed it so that the sql request only gets the users with no rooms and the matching role